liquibase-core/src/test/java/liquibase/parser/core/yaml/YamlChangeLogParserTest.java

```diff
@@ -0,0 +1,292 @@
+package liquibase.parser.core.yaml;
+
+import liquibase.changelog.ChangeLogParameters;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.exception.ChangeLogParseException;
+import liquibase.resource.ClassLoaderResourceAccessor;
+import liquibase.resource.ResourceAccessor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Unit tests for {@link YamlChangeLogParser}.
+ */
+public class YamlChangeLogParserTest {
+
+    private YamlChangeLogParser parser;
+    private ChangeLogParameters changeLogParameters;
+
+    @BeforeEach
+    void setUp() {
+        parser = new YamlChangeLogParser();
+        changeLogParameters = new ChangeLogParameters();
+    }
+
+    @Test
+    void supports_withYamlExtension_returnsTrue() {
+        ResourceAccessor resourceAccessor = new ClassLoaderResourceAccessor();
+
+        assertTrue(parser.supports("test.yaml", resourceAccessor));
+        assertTrue(parser.supports("test.yml", resourceAccessor));
+        assertTrue(parser.supports("path/to/changelog.yaml", resourceAccessor));
+        assertTrue(parser.supports("path/to/changelog.yml", resourceAccessor));
+    }
+
+    @Test
+    void supports_withNonYamlExtension_returnsFalse() {
+        ResourceAccessor resourceAccessor = new ClassLoaderResourceAccessor();
+
+        assertFalse(parser.supports("test.xml", resourceAccessor));
+        assertFalse(parser.supports("test.json", resourceAccessor));
+        assertFalse(parser.supports("test.sql", resourceAccessor));
+        assertFalse(parser.supports("test.txt", resourceAccessor));
+    }
+
+    @Test
+    void supports_withUpperCaseExtension_returnsTrue() {
+        ResourceAccessor resourceAccessor = new ClassLoaderResourceAccessor();
+
+        assertTrue(parser.supports("test.YAML", resourceAccessor));
+        assertTrue(parser.supports("test.YML", resourceAccessor));
+        assertTrue(parser.supports("test.Yaml", resourceAccessor));
+    }
+
+    @Test
+    void getPriority_default_returnsDefaultPriority() {
```

Comment:
Not sure such kind of test is useful (looks more like a change detector) - what do you think about removing it?